internal/core/src/index/VectorIndex.h

```diff
@@ -81,5 +81,5 @@ class VectorIndex : public IndexBase {
     int64_t dim_;
 };
 
-using VectorIndexPtr = std::unique_ptr<VectorIndex>;
+using VectorIndexPtr = std::unique_ptr<knowhere::Index<knowhere::IndexNode>>;
```

Comment:
knowhere::IndexNode seems to be a type of knowhre that is not exposed and should not be used directly in milvus

Reply:
And this change breaks the inheritance relationship between VectorIndexPtr and indexbase, VectorDiskannIndex, vectorMemIndex